tests/cluster/check_vagrant.py

```diff
@@ -60,3 +63,73 @@ def check_one_host_parsing(self, monkeypatch):
         assert(node2.account.user == "vagrant")
         assert(node2.account.ssh_args.strip() == "-o 'HostName 127.0.0.2' -o 'Port 2200' -o 'UserKnownHostsFile /dev/null' -o 'StrictHostKeyChecking no' -o 'PasswordAuthentication no' -o 'IdentityFile /Users/foo/ducktape.git/.vagrant/machines/worker2/virtualbox/private_key' -o 'IdentitiesOnly yes' -o 'LogLevel FATAL'")
         assert(node2.account.ssh_hostname == '127.0.0.2')
+
+    def check_cluster_file_write(self, monkeypatch):
+        self._vagrant_ssh_data = self.two_hosts
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._vagrant_ssh_config", lambda vc: (self._vagrant_ssh_data, None))
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster.is_aws", lambda vc: False)
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._externally_routable_ip", lambda vc, node_account: "127.0.0.1")
+        try:
```

Comment:
Instead of `try/except` block here, perhaps use `assert(not os.path.exists(file_path))`?